Source/SwiftLintFramework/Rules/OpeningBraceRule.swift

```diff
@@ -29,12 +43,68 @@ public struct OpeningBraceRule: Rule {
     )
 
     public func validateFile(file: File) -> [StyleViolation] {
-        let pattern = "((?:[^( ]|[\\t\\n\\f\\r (] )\\{)"
-        let excludingKinds = SyntaxKind.commentAndStringKinds()
-
-        return file.matchPattern(pattern, excludingSyntaxKinds: excludingKinds).map {
+        return file.violatingOpeningBraceRanges().map {
             StyleViolation(ruleDescription: self.dynamicType.description,
                 location: Location(file: file, offset: $0.location))
         }
     }
+
+    public func correctFile(file: File) -> [Correction] {
+        let fileRegions = file.regions()
+        let violatingRanges = file.violatingOpeningBraceRanges().filter { range in
```

Comment:
We're now doing this region check in two places: here and in `TrailingSemicolonRule`. We should add a `File.filterEnabledRange(ranges: [NSRange], forRule: Rule)` method.